src/dynarec/arm64/dynarec_arm64_660f.c

```diff
@@ -1097,21 +1112,70 @@ uintptr_t dynarec64_660F(dynarec_arm_t* dyn, uintptr_t addr, uintptr_t ip, int n
                     INST_NAME("INSERTPS Gx, Ex, Ib");
                     nextop = F8;
                     GETGX(q0, 1);
-                    if(MODREG) {
-                        GETEX(q1, 0, 1);
+                    d0 = fpu_get_scratch(dyn);
+                    VMOVQ(d0, q0);
+                    if (MODREG) {
+                        q1 = sse_get_reg(dyn, ninst, x1, (nextop & 7) + (rex.b << 3), 0);
                         u8 = F8;
-                        VMOVQDto(x1, q1, (u8>>6)&3);
+                        uint8_t count_d = (u8 >> 4) & 3;
+                        uint8_t count_s = u8 >> 6;
+                        #define GO(A) \
+                            switch (count_d) {              \
+                                case 0:                     \
+                                    VMOVeS(d0, 0, q1, A);   \
+                                    break;                  \
+                                case 1:                     \
+                                    VMOVeS(d0, 1, q1, A);   \
+                                    break;                  \
+                                case 2:                     \
+                                    VMOVeS(d0, 2, q1, A);   \
+                                    break;                  \
+                                case 3:                     \
+                                    VMOVeS(q0, 3, q1, A);   \
+                                    break;                  \
+                            }
+                        switch (count_s) {
+                            case 0:
+                                GO(0)
+                                break;
+                            case 1:
+                                GO(1)
+                                break;
+                            case 2:
+                                GO(2)
+                                break;
+                            case 3:
+                                GO(3)
+                                break;
+                        }
+                        #undef GO
                     } else {
                         SMREAD();
-                        addr = geted(dyn, addr, ninst, nextop, &wback, x1, &fixedaddress, &unscaled, 0xfff<<2, 3, rex, NULL, 0, 1);
-                        LDW(x1, wback, fixedaddress);
+                        addr = geted(dyn, addr, ninst, nextop, &ed, x1, &fixedaddress, &unscaled, 0xfff<<4, 3, rex, NULL, 0, 0);

```

Comment:
Same here, you need that delta of `1`

Reply:
Also, the `0xfff<<4` is not coherent with a mask of `3`. You are using a `LDW` opcode, wich is a 32bits read, so it should be `0xfff<<2` and mask `3`